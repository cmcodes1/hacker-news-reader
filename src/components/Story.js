import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import dateFromNow from '../helpers/DateFromNow';
import colors from '../theme/colors';

export default function Story({ navigation, index, title, url, score, by, time, descendants, kids, }) {

    const getHostname = (url) => {
        const matches = url.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
        return matches && matches[1];
    }

    return (
        <View style={styles.item}>
            <Text style={styles.row}>
                <Text style={styles.bigText}>{index + 1}{'. \u25B2 '}</Text>
                <Text style={styles.title}>{title}</Text>
                <Text style={styles.bigText}>{` (${getHostname(url)})`}</Text>
            </Text>
            <Text style={styles.row}>
                <Text style={styles.smallText}>{score}{' points by '}{by}{' '}{dateFromNow(time)}</Text>
                <Text
                    style={styles.smallText}
                    onPress={() => navigation.navigate('Item', { kids: kids })}
                >
                    {descendants && ` | ${descendants} comments`}
                </Text>
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    item: {
        margin: 10,
    },
    row: {
        flexDirection: 'row',
        flexWrap: 'wrap',
    },
    title: {
        fontSize: 14,
        color: colors.textPrimary,
    },
    bigText: {
        fontSize: 14,
    },
    smallText: {
        fontSize: 11,
    },
})