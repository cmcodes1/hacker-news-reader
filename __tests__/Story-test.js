import React from 'react'
import { create } from 'react-test-renderer'
import Story from '../src/components/Story'

jest.useFakeTimers()

const tree = create(<Story />)

test('snapshot', () => {
    expect(tree).toMatchSnapshot()
})