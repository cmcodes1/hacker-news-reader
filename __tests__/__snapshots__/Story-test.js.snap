// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      NaN
      . â–² 
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    />
  </Text>
  <Text
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 11,
        }
      }
    >
       points by 
       
      on undefined at 12:NaN am
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 11,
        }
      }
    />
  </Text>
</View>
`;
